sources/api/netdog/src/cli/mod.rs

```diff
@@ -47,9 +48,50 @@ where
     Ok(())
 }
 
+/// Return the primary interface name
+// A primary_interface or primary_mac_address file should exist.  If the primary_interface file
+// exists use it, otherwise read the primary_mac_address file and crawl the leases to find which
+// interface has the corresponding MAC, if any.
+fn primary_interface_name() -> Result<String> {
+    let clean = |s: String| s.trim().to_lowercase();
+
+    let maybe_name = fs::read_to_string(PRIMARY_INTERFACE).ok();
+    if let Some(name) = maybe_name {
+        return Ok(clean(name));
+    }
+
+    let primary_mac = clean(fs::read_to_string(PRIMARY_MAC_ADDRESS).context(
+        error::PathReadSnafu {
+            path: PRIMARY_MAC_ADDRESS,
+        },
+    )?);
+    let leases = fs::read_dir(LEASE_DIR)
+        .context(error::PathReadSnafu { path: LEASE_DIR })?
+        .flatten()
+        .filter(|p| {
+            p.path().is_file()
+                && p.file_name()
+                    .to_str()
+                    .map_or(false, |f| f.starts_with("leaseinfo"))
+        });
+
+    for lease in leases {
+        let lease_info =
+            LeaseInfo::from_lease(lease.path()).context(error::LeaseParseFailedSnafu)?;
+
+        if let Some(lease_mac) = lease_info.mac_address {
+            if primary_mac == clean(lease_mac) {
+                return Ok(clean(lease_info.interface_name));
+            }
+        }
```

Comment:
I have some misgivings about this:
* it seems like we're reading lease files while being actively called as a helper by wicked, and therefore we might potentially read files in an inconsistent state 
* overall there's a fair amount of parsing required and we have to do it on every lease event, which can happen frequently for IPv6
* we end up reading and parsing the same lease file twice, since it's read again by `fetch_lease`

Can we iterate over `/sys/class/net/*/address` instead and return the name of the directory where the address file matches?